src/renderer/services/chain/fees/deposit.ts

```diff
@@ -60,64 +57,136 @@ export const reloadFees$: Rx.Observable<O.Option<LoadFeesHandler>> = selectedPoo
-        RxOp.switchMap((oMemo) =>
-          FP.pipe(
-            oMemo,
-            O.fold(
-              () => Rx.of(RD.initial),
-              (memo) => BTC.feesWithRates$(memo).pipe(liveData.map(({ fees }) => fees.fast))
-            )
-          )
+      return FP.pipe(
+        memo,
+        O.fold(
+          () => Rx.of(RD.initial),
+          (memo) => BTC.feesWithRates$(memo).pipe(liveData.map(({ fees }) => fees.fast))
         )
       )
-    case THORChain:
+    }
+
+    case THORChain: {
       return THOR.fees$().pipe(liveData.map(({ fast }) => fast))
-    case ETHChain:
-      return Rx.of(RD.failure(Error('Deposit fee for ETH has not been implemented')))
+    }
+    case ETHChain: {
+      return FP.pipe(
+        sequenceSOption({ recipient, amount, memo }),
+        O.fold(
+          () => Rx.of(RD.initial),
+          ({ recipient, amount, memo }) =>
+            ETH.fees$({ recipient, amount, memo }).pipe(liveData.map((fees) => fees.fast))
+        )
+      )
+    }
     case CosmosChain:
       return Rx.of(RD.failure(Error('Deposit fee for Cosmos has not been implemented')))
     case PolkadotChain:
       return Rx.of(RD.failure(Error('Deposit fee for Polkadot has not been implemented')))
-    case BCHChain:
-      return BCH.fees$().pipe(liveData.map(({ fast }) => fast))
-    case LTCChain:
-      return LTC.fees$().pipe(liveData.map(({ fast }) => fast))
+    case BCHChain: {
+      // deposit fee for BCH txs based on a memo,
+      // and memo depends on deposit type
+      return FP.pipe(
+        memo,
+        O.fold(
+          () => Rx.of(RD.initial),
+          (memo) => BCH.feesWithRates$(memo).pipe(liveData.map(({ fees }) => fees.fast))
+        )
+      )
+    }
+    case LTCChain: {
+      // deposit fee for LTC txs based on a memo,
+      // and memo depends on deposit type
+      return FP.pipe(
+        memo,
+        O.fold(
+          () => Rx.of(RD.initial),
+          (memo) => LTC.feesWithRates$(memo).pipe(liveData.map(({ fees }) => fees.fast))
+        )
+      )
+    }
   }
 }
 
-const depositFees$ = (type: DepositType): DepositFeesLD =>
+const depositFees$ = (initialParams: DepositFeesParams): DepositFeesLD =>
   FP.pipe(
-    Rx.combineLatest([selectedPoolChain$, reloadDepositFees$]),
-    RxOp.switchMap(([oPoolChain]) =>
+    reloadDepositFees$,
+    RxOp.debounceTime(300),
+    RxOp.distinctUntilChanged((prev, next) => {
+      if (prev && next) {
+        /**
+         * Pass values only in cases when:
+         * 1 - chain was changed
+         * 2 - Amount to deposit was changed. Some chains' fess depends on amount too (e.g. ETH)
+         */
+        return (
+          // Check if entered chain was changed
+          eqChain.equals(prev.asset.chain, next.asset.chain) &&
+          // Check if entered amount was changed
+          eqBaseAmount.equals(prev.amount, next.amount)
+        )
+      }
+      return false
+    }),
```

Comment:
How about to provide a `eqDepositFeesParams` helper?

Reply:
yeah. no problem. just as it used in a single place have not done it. will add it in few minutes